### ISSUE TITLE
Preventing display of input data

### ISSUE THREAD

**Issue by aubreyz** [bwanders/dokuwiki-strata#31]
There appears to be no way to do this within the existing plugin, even though it is achieved on this website

http://www.fwurg.net/dokuwiki/

I get a great big horrible table with each block of input code (making the plugin useless to me).
Is this a bug, or is there something missing from the manual?

None of the configuration options seem relevant to not-displaying every input block as a table, even though this seems possible.

**Comment by bwanders** [comment]
Well, changing the appearance of the data entry blocks can be done through the use of the https://github.com/bwanders/dokuwiki-plugin-templatery and https://github.com/bwanders/dokuwiki-plugin-stratatemplatery plugins.

You can then setup up a template to display only some (or none) of the information from the entry block. This can be seen in action at http://www.fwurg.net/dokuwiki/template:guide. If you need help with any of my other plugins, do not hesitate to open an issue on them!

So, to answer the direct question: this is not a bug, it's a combination of multiple plugins.

Does this help out?

(Maybe they should feature more prominently in the readme, and the templatery and stratatemplatery plugins ought to have a dokuwiki plugin page... But stratatemplatery strongly depends on both strata and templatery, and this cannot really be expressed on the dokuwiki site, so I'm hesitant since it will lead to broken wiki's and sad/angry people.)

**Comment by aubreyz** [comment]
Thank you Brend (and for all your coding work). OK, this looks like an extremely useful linked sets of plugins. I am going to have a try. So can I probe you slightly about future bugs. Strata looks like the most useful semantic markup method in DW by a long way. What I don't want to do is develop a complex system based on interlinked plugins and then find that the whole things falls over permanently in 3 years when PHP or DW is upgraded. Perhaps give us a heads up on your plans...

Can I also suggest that (perhaps) the ability to simply turn off data input display completely is so fundamental  that at least that aspect should be included in Strata itself.

**Comment by bwanders** [comment]
### On the ability to turn off data input display

You already have this option! All data entries are neatly built up in HTML with the `strata-entry` class on the entry element. So you can use the [userstyle](https://www.dokuwiki.org/devel:css#user_styles) mechanism. A simple rule like:
```css
.dokuwiki .strata-entry {
    display: none;
}
```
Would achieve what you propose.

Alternatively, wrapping those data blocks that need to be hidden in either a custom style with the [wrap plugin](https://www.dokuwiki.org/plugin:wrap) (the wrap plugin is almost a requirement for any wiki that wishes to do even the most basic of information display that is not based on flowing text).

Another strong option would be to use [stratainline](https://github.com/bwanders/dokuwiki-plugin-stratainline) for inlining a lot of the data. In my experience, this is usually enough to get the amount of data down to a managable level, without hiding things from the user directly -- I have found that most users are quicker to pick up things that have a concrete presence, and can be seen to work.

### On long-term plans for Strata

I fully understand your desire to have long-term support for the systems you use. I have been using and developing the Strata plugins for a little over five years now, and will continue to develop and update them as long as I continue using them. However, I have been drifting away from actively developing for Strata, and are mostly a user of the system right now. I foresee at least another two years of continued use in the future, but dare not say anything beyond that point.

PHP is in the unfortunate situation that there are semantic changes that ought to be made in the language (and PHP will be better for it) but when such changes are made they subtly break older code in unexpected ways. PHP is not one of my main programming languages, and this makes it somewhat time-expensive to track down and fix these issues. Since I do this in my spare time now, I have to ration what time I have to work on this and other projects. 

(Of course the whole Strata plugin structure is open-sourced, so the projects will not suddenly disappear on you. It's just that I can not guarantee long-term support.)

tl;dr: I will try my best to keep everything running smoothly, but can not guarantee continued updates over long-term periods.
